gammapy/data/observations.py

```diff
@@ -491,7 +542,9 @@ def write(self, path, overwrite=False, format="gadf", include_irfs=True):
 
         events = self.events
         if events is not None:
-            hdul.append(events.to_table_hdu(format=format))
+            events_hdu = events.to_table_hdu(format=format)
```

Comment:
I have a question that might be related to an other PR.
As for each 'format' the versions can evolve, should be 'extended' the parameters list of the `to_table_hdu` function with a version ID?
The version of the format is written into the DL3 meta data and this information should be kept into the internal meta data such that it can be re-used.

Reply:
Yes, we should

Reply:
Relevent issue #4239